### DISCUSSION TITLE
Show Originals: the all-posts-are-hidden problem

### DISCUSSION THREAD

**Discussion by marcustyphoon** [AprilSylph/XKit-Rewritten#107] [Tech Office]
As has been discussed at length back when No Recommended was a thing in XKit 7, it's possible for post-hiding extensions to hide so many posts that the dashboard has no scrollbar, which causes infinite scrolling not to be triggerable. Show Originals can cause this to happen. (Apparently Tumblr is showing me only 4 posts in the initial page load now, I guess...? That makes it happen a fair amount.)

The method I used to fix this in my XKit 7 PR was to put a `min-height: 100vh` on the timeline div, guaranteeing that it extends off the end of the screen and always creates a scroll bar. (I actually made it only extend 10px below the screen edge via `calc(100vh - ${$(timelineSel).offset().top - 10}px)`, though that may be a bit overcomplicated for not much benefit.)

I can't think of a major side effect of this, besides looking a little strange if your dash somehow has ~no posts and your browser window is really tall, but it feels kind of hacky and maybe someone has an idea for a better solution?

**Comment by AprilSylph** [comment]
doesn't sound too hacky to me! "always allow the user to scroll" sounds simple and reasonable, much more so than scripts attempting to trigger post loading automatically. it would need to check for infinite scroll being enabled, though; setting that style on timeline elements when the user has dashboard pagination enabled would make no sense at all

**Reply by marcustyphoon** [reply]
Yep, I checked for the presence of a "next" button before applying the style.

The biggest thing making it feel like a hack to me is scrolling all the way to the bottom, causing an increase in the number of visible posts that... still fits on your display, requiring you to scroll _up_ to load more. I think this is pretty minor and all my experimentation with booping the height by 1px to simulate a scroll is far more trouble than it's worth.

Can't wait for that data-timeline attribute... :D
